File: Python/2-locating-elements/6-By-Partial-Link.py
from selenium import webdriver

driver = webdriver.Chrome()

try:
    driver.get('https://www.selenium.dev/')

    '''
    Fetch the all the Link DOM node with text containing "DOWNL"
    '''
    my_links = driver.find_elements_by_partial_link_text("DOWNL")
    for link in my_links:
        # Access the attributes value
        print(link.get_attribute('href'))

finally:
    driver.quit()

File: Javascript/2-locating-elements/6-By-Partial-Link.js
const {Builder, By} = require('selenium-webdriver');

(async function(){
    const driver = new Builder().forBrowser('chrome').build();

    try {
        (await driver).get("https://www.selenium.dev/")

        /**
         * Fetch the all the Link DOM node with text containing "DOWNL"
         */
        
        let my_links = (await driver).findElements(By.partialLinkText("DOWNL"));
        for (let my_link of await my_links){
            let my_href = (await my_link).getAttribute('href');
            console.info(await my_href);
        }

    } finally {
        (await driver).quit();
    }

})();